runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/StaticBundleFactory.java

```diff
@@ -0,0 +1,14 @@
+package org.apache.beam.runners.fnexecution.control;
+
+import org.apache.beam.runners.fnexecution.artifact.ArtifactSource;
+import org.apache.beam.runners.fnexecution.provisioning.JobInfo;
+
+/**
+ * A factory that can create {@link JobBundleFactory JobBundleFactories} from job-scoped resources
+ * provided by an operator.
+ *
+ * <p>This should be a global or singleton resource that caches JobBundleFactories between calls.
```

Comment:
Is the `StaticBundleFactory` itself meant to be a global singleton or is the retuned result meant to be singleton scoped (per job)? This might be clearer as a class rather than interface.

Reply:
Yeah I think this doesn't need an interface. I'm also not sure whether its implementation per se should be runner-specific, or whether one should pass runner-specific hooks into a general-purpose implementation. For this PR, probably fine to either convert this to a class with a static method, or omit.

Reply:
removed